api_intro.py

```diff
@@ -21,75 +28,63 @@ def get_access_token(key, secret):
-    response = request.json()
-    return response["data"]["attributes"]["keywords"]
 
-if __name__ == "__main__":
-    ob_access_token = get_access_token(ob_client_id, ob_client_Secret)
-    sn_access_token = get_access_token(sn_client_id, sn_client_Secret)
+if __name__ == '__main__':
+    ob_access_token = get_access_token(OB_CLIENT_ID, OB_CLIENT_SECRET)
+    sn_access_token = get_access_token(SN_CLIENT_ID, SN_CLIENT_SECRET)
 
     # API Call 1: Service Now API Retrieve Top 5 College Data
-    print("API Call 1: Service Now API Retrieve College Data")
-    collegeData = get_svcnow_colleges(sn_access_token)
-    count = 1
-    for college in collegeData:
-        print(str(count) + "- College Code: " + college["attributes"]["code"] + ", College Name: " + college["attributes"]["description"])
-        count += 1
-        if count > 5:
-            break
-    
+    print('API Call 1: Service Now API Retrieve College Data')
+    college_data = get_api_response(sn_access_token, svcnow_url)
+    for college in college_data[:5]:
+        print(f"- College Code: {college['attributes']['code']}" +
+              f", College Name: {college['attributes']['description']}")
+
     # API Call 2: Get Onbase Documents
-    print("API Call 2: Retrieve Onbase Documents")
-    documentData = get_onbase_documents(ob_access_token)
-    count = 1
-    for document in documentData:
-        print(str(count) + "- Doc ID: " + document["id"] + ", Doc Name: " + document["attributes"]["name"] + ", Created Date: " + document["attributes"]["storedDate"])
-        keywords = get_onbase_keywords(ob_access_token, document["id"])
+    print('API Call 2: Retrieve Onbase Documents')
+    end_date = str(date.today())
+    url_filter = (f'?filter[documentTypeName]=ADMS Deny&filter[endDate]=' +
+                  f'{end_date}')
+    document_data = get_api_response(ob_access_token, onbase_url + url_filter)
+    for document in document_data[:5]:
+        print(f"- Doc ID: {document['id']}, Doc Name: " +
+              f"{document['attributes']['name']}, Created Date: " +
+              f"{document['attributes']['storedDate']}")
+        doc_url = f"/{document['id']}/keywords"
+
+        # API Call 3: Get Onbase Document Keywords
+        keyword_data = get_api_response(ob_access_token, onbase_url + doc_url)
+        keywords = keyword_data['attributes']['keywords']
         for keyword in keywords:
-            if keyword["name"] == 'ADMS - Application Term':
-                print("-- Application Term: " + keyword["values"][0]["formattedValue"])
-            if keyword["name"] == 'ADMS - App Number':
-                print("-- Application Number: " + str(keyword["values"][0]["formattedValue"]))
-        count += 1
-        if count > 5:
-            break
\ No newline at end of file
+            if keyword['name'] == 'ADMS - Application Term':
+                print('-- Application Term: ' +
+                      f"{keyword['values'][0]['formattedValue']}")
```

Comment:
Is it possible `keyword['values']` can be an empty array?

Reply:
They shouldn't on these documents, but I added a check for this just in case.